src/integrations/__tests__/unixIntegrationManager.spec.ts

```diff
@@ -47,287 +43,260 @@ afterEach(async() => {
   }
 });
 
-testUnix('Ensure symlinks and dirs are created properly', async() => {
-  const integrationManager = new UnixIntegrationManager(
-    resourcesDir, integrationDir, dockerCliPluginDir);
+describeUnix('UnixIntegrationManager', () => {
+  let integrationDir: string;
+  let dockerCliPluginDir: string;
+  let integrationManager: UnixIntegrationManager;
 
-  await integrationManager.enforce();
-  expect(fs.promises.readdir(integrationDir)).resolves.not.toThrow();
-  for (const name of await fs.promises.readdir(resourcesDir)) {
-    const integrationPath = path.join(integrationDir, name);
+  beforeEach(() => {
+    integrationDir = path.join(testDir, INTEGRATION_DIR_NAME);
+    dockerCliPluginDir = path.join(testDir, 'dockerCliPluginDir');
+    integrationManager = new UnixIntegrationManager(
+      resourcesDir, integrationDir, dockerCliPluginDir);
+  });
 
-    expect(fs.promises.readlink(integrationPath, 'utf8')).resolves.not.toThrow();
-  }
-  for (const name of await integrationManager.getDockerCliPluginNames()) {
-    const pluginPath = path.join(dockerCliPluginDir, name);
+  test('.enforce() should create dirs and symlinks properly', async() => {
+    await integrationManager.enforce();
+    expect(fs.promises.readdir(integrationDir)).resolves.not.toThrow();
+    for (const name of await fs.promises.readdir(resourcesDir)) {
+      const integrationPath = path.join(integrationDir, name);
 
-    expect(fs.promises.readlink(pluginPath, 'utf8')).resolves.not.toThrow();
-  }
-});
+      expect(fs.promises.readlink(integrationPath, 'utf8')).resolves.not.toThrow();
```

Comment:
Also we typically don't use `resolves.not.toThrow()` -- more like we assert on what we expect to see. The output is probably an empty string, so `resolves.toBe('')` is probably better.

Reply:
I did the same thing here (more or less) as you suggested here: https://github.com/rancher-sandbox/rancher-desktop/pull/3183/files/242bcdc669e33c75021d6956bc8e871860d75f6c#r998655706